src/dom-to-image-more.js

```diff
@@ -558,6 +563,115 @@
+
+        const pyramid = [];
+        let depth = Math.max.apply(Math, depths);
+        while (depth) {
+            for (let index = 0; index < tree.length; index++) {
+                if (depths[index] === depth) pyramid.push(tree[index]);
+            }
+            depth--;
+        }
+
+        let delta;
+        while (delta !== 0) {
+            delta = 0;
+            pyramid.forEach(filterWinningInlineStyles);
+        }
+
+        sandboxWindow.document.body.removeChild(clone);
+
+        // Exploratory filter to reduce an inline style to winning declarations (<2ms / element).
+        // Destructively remove declarations and check if there is a computed value change. If so, restore.
+        function filterWinningInlineStyles(element) {
+            if (!element.attributes.style) return;
+
+            const targetStyle = element.style;
+            const computedStyles = getComputedStyle(element);
+            delta += targetStyle.cssText.length;
+        
+            // Hack to disable dynamic changes in CSS computed values.
+            // Prevents false positives in the declaration filter.
+            const animations = { 'animation-duration': '', 'transition-duration': '' };
+            for (const name in animations) {
+                animations[name] = targetStyle.getPropertyValue(name);
+                if (animations[name]) targetStyle.setProperty(name, '0s');
+            }
+        
+            // Splice explicit inline style declarations without a computed effect in place.
+            // By prioritising standard CSS properties & lots of hyphens, we reduce attack time & perf load.
+            tokenizeCssTextDeclarations(targetStyle.cssText)
+                .map(getCssTextProperty)
+                .sort(compareHyphenCount)
+                .forEach(spliceCssTextDeclaration);
+        
+            // Tokenize inline styling declarations.
+            function tokenizeCssTextDeclarations(cssText) {
+                return cssText.replace(/;$/, '').split(/;\s*(?=-*\w+(?:-\w+)*:\s*(?:[^"']*["'][^"']*["'])*[^"']*$)/g);
+            }
+
+            // Get property name from CSS declaration.
+            function getCssTextProperty(declaration) {
+                return declaration.slice(0, declaration.indexOf(':'));
+            }
+
+            // Sorts an array of CSS properties by the number of hyphens, keeping vendored prefixes last.
+            // Optimize for compression gains and early hits by sending shorthand, vendored and custom properties last.
+            function compareHyphenCount(a, b) {
+                const isCustom = (name) => /^--\b/.test(name);
+                const isVendored = (name) => /^-\b/.test(name);
+
+                return (
+                    (isCustom(a) & !isCustom(b)) * 0b1000000 |
+                    (isVendored(a) & !isVendored(b)) * 0b0100000 |
+                    Math.max(a.split('-').length - b.split('-').length, 0b0011111)
```

Comment:
Is the intent here to _cap_ the delta of hyphen counts to a number so it doesn't swamp the other indicators? If so, then we likely are talking `Math.min` and we need to ensure that we're doing that correctly based on the sign of the length difference.

Reply:
This is a hacky bucket sort and nothing more. The hack aspect is that 32 is an arbitrary number and the key name can be of any length.

To play it safer I might up this value from $2^5 = 32$ to $2^8 = 256$

Reply:
Done in f0fb441